### PR TITLE
[ENH] Implemented CBLOF for Anomaly Detection

### LINKED ISSUE #2110: [ENH] Add PyODAdapter-implementation for CBLOF
### Describe the feature or idea you want to propose

The [`PyODAdapter`](https://github.com/aeon-toolkit/aeon/blob/main/aeon/anomaly_detection/_pyodadapter.py) in aeon allows us to use any outlier detector from [PyOD](https://github.com/yzhao062/pyod), which were originally proposed for relational data, also for time series anomaly detection (TSAD). Not all detectors are equally well suited for TSAD, however. We want to represent the frequently used and competitive outlier detection techniques within the `anomaly_detection` module of aeon directly.

**Implement the [CBLOF method](https://github.com/yzhao062/pyod/blob/master/pyod/models/cblof.py#L25)** using the `PyODAdapter`.

### Describe your proposed solution

- Create a new file in `aeon.anomaly_detection` for the method
- Create a new estimator class with `PyODAdapter` as the parent
- Expose the algorithm's hyperparameters as constructor arguments, create the PyOD model and pass it to the super-constructor
- Document your class
- Add tests for certain edge cases if necessary

---

Example for IsolationForest:

```python
class IsolationForest(PyODAdapter):
    """documentation ..."""
    def __init__(n_estimators: int = 100, max_samples: int | str = "auto", ..., window_size: int, stride: int):
        model = IForest(n_estimators, max_samples, ...
        super().__init__(model, window_size, stride)

    @classmethod
    def get_test_params(cls, parameter_set="default"):
        """..."""
        return {"n_estimators": 10, ...}
```

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by notaryanramani**
#### Reference Issues/PRs

Fixes #2110 

#### What does this implement/fix? Explain your changes.

Implements CBLOF for Anomaly Detection

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [x] I've added the estimator to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

File: aeon/anomaly_detection/tests/test_cblof.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    copod = CBLOF(window_size=1, stride=1, random_state=2)
+    copod_preds = copod.fit_predict(series)
+
+    # fit and predict with PyOD CBLOF
+    _series = series.reshape(-1, 1)
+    pyod_copod = PyODCBLOF(random_state=2)
+    pyod_copod.fit(_series)
+    pyod_copod_preds = pyod_copod.decision_function(_series)
+
+    assert np.allclose(copod_preds, pyod_copod_preds)
```
Comment (CodeLionX) [inline aeon/anomaly_detection/tests/test_cblof.py:67]: Please use `np.testing.assert_allclose` for the better error message

**Review by CodeLionX — CHANGES_REQUESTED** [review]
Good work 👍🏼 

Because CBLOF uses a clustering estimator internally, I would like to see an additional test that passing a custom estimator instance also works.

Do you think that it makes sense to also test with a time series clusterer from aeon? IMO, we use the sliding window approach to transform the time series into another space, where traditional metrics can be used. So, using `TimeSeriesKMeans` might not benefit the approach 🤷🏼

**Comment by notaryanramani** [comment]
@CodeLionX 

> Because CBLOF uses a clustering estimator internally, I would like to see an additional test that passing a custom estimator instance also works.

Okay, will add a test that uses another estimator.

> Do you think that it makes sense to also test with a time series clusterer from aeon? IMO, we use the sliding window approach to transform the time series into another space, where traditional metrics can be used. So, using `TimeSeriesKMeans` might not benefit the approach 🤷🏼

I think we can add a test to check if `CBLOF` can also be used with an aeon clustering estimator or to check the API consistency.

**Review by CodeLionX — APPROVED** [review]

**Comment by notaryanramani** [comment]
@CodeLionX 

`COPOD` requies the shape of cluster centers to be 2d to calculate distance, but cluster centers for `TimeSeriesKMeans` is 3d. Is there a way for cluster centers to be 2d if n_channels is 1?

**Comment by CodeLionX** [comment]
> @CodeLionX
> 
> `COPOD` requies the shape of cluster centers to be 2d to calculate distance, but cluster centers for `TimeSeriesKMeans` is 3d. Is there a way for cluster centers to be 2d if n_channels is 1?

In this case, I would just state in the documentation that Aeon estimators are not supported. It does not make much sense to use them anyway.

**Review by CodeLionX — APPROVED** [review]

**Comment by TonyBagnall** [comment]
thanks for the contribution
